### PR TITLE
add a wallet merge tool

### PR DESCRIPTION
**Pull request by DavidVorick**
There is now a command to help you merge wallets together, which means you're not at risk of losing coins when deleting or moving around other wallet files. This is also useful for people mining on many computers at the same time.

Thanks to this feature, I was able to recover 4m of the 6m I lost. I now am not sure whether I actually lost 6m or just 4m. Regardless, a significant chunk is back.

### CODE REVIEW

File: modules/wallet/persist.go [hunk 1]
```diff
@@ -129,3 +137,19 @@ func (w *Wallet) load() error {
...
+	err := encoding.ReadFile(filepath, &savedKeys)
+	if err != nil {
+		return err
+	}
+	err = w.loadKeys(savedKeys)
+	if err != nil {
+		return err
+	}
+	return nil
+}
```
Comment (lukechampine) [inline modules/wallet/persist.go]: seems to me like "Merge" is really just "load"

Comment (DavidVorick) [inline modules/wallet/persist.go]: initially it wasn't, but I think I changed load to work like Merge.

`load` also grabs outputs, but I can deduplicate a lot of this.

**Comment by lukechampine** [comment]
how does this relate to eventual loading of multiple wallets?
would wallet merging be strictly superseded by multiple wallet support?

**Comment by DavidVorick** [comment]
I believe it would

**Comment by DavidVorick** [comment]
rebased to master, cleaned files up, and then fixed a siafund reporting bug.

File: modules/wallet/persist.go [hunk 2]
```diff
@@ -45,12 +38,29 @@ func (w *Wallet) save() error {
 	// 'siafunds.dat' or something similar, people might think it's okay to
 	// delete their siafund keys, which is NOT okay. Instead of potentially
 	// having this confusion, I chose a less suggestive name.
-	err = encoding.WriteFile(filepath.Join(w.saveDir, "outputs.backup"), siafundSlice)
+	return encoding.WriteFile(filepath, siafundSlice)
```
Comment (lukechampine) [inline modules/wallet/persist.go]: need to move the comment

**Comment by DavidVorick** [comment]
done

**Comment by lukechampine** [comment]
I don't like implementing API calls that we know will soon be deprecated. Do we even plan on supporting this in UI? If not, why not just confine it so siac?

**Comment by DavidVorick** [comment]
not really, but this is very important for the people who have lost funds in the same way that I did.

Also, it really eases some of the upgrading tension.

I feel strongly about this PR. Right now there are lots of people (especially multi-machine people) who would like some way to combine their wallets.

**Pull request merged**